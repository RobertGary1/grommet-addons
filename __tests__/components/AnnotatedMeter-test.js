// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import AnnotatedMeter from '../../src/js/components/AnnotatedMeter';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('AnnotatedMeter', () => {
  it('has correct simple options - bar', () => {
    const component = renderer.create(
      <AnnotatedMeter type='bar' series={[
        { label: 'label-1', value: 10 },
        { label: 'label-2', value: 20 }
      ]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
    it('has correct simple options - circle', () => {
    const component = renderer.create(
      <AnnotatedMeter type='circle' series={[
        { label: 'label-1', value: 10 },
        { label: 'label-2', value: 20 }
      ]}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
