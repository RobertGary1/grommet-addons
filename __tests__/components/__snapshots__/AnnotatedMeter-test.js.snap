exports[`AnnotatedMeter has correct simple options - bar 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <div
      className="grommetux-box grommetux-box--direction-row grommetux-box--justify-between grommetux-box--align-center grommetux-box--pad-between-small"
      id={undefined}
      onClick={undefined}
      role={undefined}
      style={Object {}}
      tabIndex={undefined}>
      <div
        className="grommetux-value grommetux-value--align-start">
        <div
          className="grommetux-value__annotated">
          <div>
            <span
              className="grommetux-value__value">
              30
            </span>
          </div>
        </div>
      </div>
      <span>
        <span>
          Total
        </span>
      </span>
    </div>
    <div
      className="grommetux-meter grommetux-meter--bar grommetux-meter--stacked grommetux-meter--count-2">
      <div
        className="grommetux-meter__value-container">
        <div
          className="grommetux-meter__graphic-container">
          <svg
            aria-label="Bar Meter. , Value: 30, Max: 100"
            className="grommetux-meter__graphic"
            height={24}
            onBlur={[Function]}
            onFocus={[Function]}
            preserveAspectRatio="xMidYMid meet"
            role="group"
            tabIndex="0"
            viewBox="0 0 192 24"
            width={192}>
            <g
              className="grommetux-meter__tracks">
              <g>
                <path
                  className="grommetux-meter__slice"
                  d="M0,12 L192,12"
                  data-index={0}
                  onBlur={[Function]}
                  onFocus={[Function]} />
                <path
                  className="grommetux-meter__hot"
                  d="M0,12 L192,12"
                  fill="none"
                  onClick={undefined}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]} />
              </g>
            </g>
            <g
              className="grommetux-meter__values">
              <g
                aria-label="10"
                role="img">
                <path
                  className="grommetux-meter__slice grommetux-color-index-graph-1"
                  d="M0,12 L20,12"
                  data-index={0}
                  onBlur={[Function]}
                  onFocus={[Function]} />
                <path
                  className="grommetux-meter__hot"
                  d="M0,12 L20,12"
                  fill="none"
                  onClick={undefined}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]} />
              </g>
              <g
                aria-label="20"
                role="img">
                <path
                  className="grommetux-meter__slice grommetux-color-index-graph-2"
                  d="M20,12 L59,12"
                  data-index={1}
                  onBlur={[Function]}
                  onFocus={[Function]} />
                <path
                  className="grommetux-meter__hot"
                  d="M20,12 L59,12"
                  fill="none"
                  onClick={undefined}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]} />
              </g>
            </g>
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AnnotatedMeter has correct simple options - circle 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--align-start grommetux-box--responsive grommetux-box--pad-none"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-none"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}>
    <div
      className="grommetux-meter grommetux-meter--circle grommetux-meter--stacked grommetux-meter--count-2">
      <div
        className="grommetux-meter__value-container">
        <div
          className="grommetux-meter__graphic-container">
          <svg
            aria-label="Circle Meter. , Value: 30, Max: 100"
            className="grommetux-meter__graphic"
            height={192}
            onBlur={[Function]}
            onFocus={[Function]}
            preserveAspectRatio="xMidYMid meet"
            role="group"
            tabIndex="0"
            viewBox="0 0 192 192"
            width={192}>
            <g
              className="grommetux-meter__tracks">
              <g>
                <path
                  className="grommetux-meter__slice"
                  d="M 96.0146607656 179.9999987206 A 84.0000000000 84.0000000000 0 1 0 96.0000000000 180.0000000000"
                  data-index={0}
                  onBlur={[Function]}
                  onFocus={[Function]} />
                <path
                  className="grommetux-meter__hot"
                  d="M 96.0146607656 179.9999987206 A 84.0000000000 84.0000000000 0 1 0 96.0000000000 180.0000000000"
                  fill="none"
                  onClick={undefined}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]} />
              </g>
            </g>
            <g
              className="grommetux-meter__values">
              <g
                aria-label="10"
                role="img">
                <path
                  className="grommetux-meter__slice grommetux-color-index-graph-1"
                  d="M 46.6260388074 163.9574275275 A 84.0000000000 84.0000000000 0 0 0 96.0000000000 180.0000000000"
                  data-index={0}
                  onBlur={[Function]}
                  onFocus={[Function]} />
                <path
                  className="grommetux-meter__hot"
                  d="M 46.6260388074 163.9574275275 A 84.0000000000 84.0000000000 0 0 0 96.0000000000 180.0000000000"
                  fill="none"
                  onClick={undefined}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]} />
              </g>
              <g
                aria-label="20"
                role="img">
                <path
                  className="grommetux-meter__slice grommetux-color-index-graph-2"
                  d="M 16.1112526312 70.0425724725 A 84.0000000000 84.0000000000 0 0 0 46.6260388074 163.9574275275"
                  data-index={1}
                  onBlur={[Function]}
                  onFocus={[Function]} />
                <path
                  className="grommetux-meter__hot"
                  d="M 16.1112526312 70.0425724725 A 84.0000000000 84.0000000000 0 0 0 46.6260388074 163.9574275275"
                  fill="none"
                  onClick={undefined}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]} />
              </g>
            </g>
          </svg>
        </div>
        <div
          className="grommetux-meter__label">
          <div
            className="grommetux-value grommetux-value--align-center">
            <div
              className="grommetux-value__annotated">
              <div>
                <span
                  className="grommetux-value__value">
                  30
                </span>
              </div>
            </div>
            <span
              className="grommetux-value__label">
              <span>
                Total
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
